### PR TITLE
Function combineStrata

### LINKED ISSUE #505: consider combineStrata function
- investigate if there exist anything similar in the tidyverse

### PR DESCRIPTION
**Pull request by KimLopezGuell**
Close #505

### CODE REVIEW

File: R/combineStrata.R [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#' @return A vector of all combinations of strata.
+#' @importFrom utils combn
+#' @export
+combineStrata <- function(levels) {
+  # Checks
+  omopgenerics::assertCharacter(levels)
+  if(length(levels) < 2) {cli::cli_abort("strata vector must have at least two levels.")}
+
+  # Apply combn function to all lengths of combinations
+  result <- lapply(c(1:length(levels)), combn, x = levels, simplify = FALSE) |>
```
Comment (catalamarti) [inline R/combineStrata.R:30]: I like more purrr: although it is the same hahahaha
purrr::map(seq_along(levels), \(x) combn(x = levels, m = x, simplify = FALSE))

File: R/combineStrata.R [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#'
+#' @param levels Vector of all strata levels to combine.
+#'
+#' @return A vector of all combinations of strata.
+#' @importFrom utils combn
+#' @export
+combineStrata <- function(levels) {
+  # Checks
+  omopgenerics::assertCharacter(levels)
+  if(length(levels) < 2) {cli::cli_abort("strata vector must have at least two levels.")}
```
Comment (catalamarti) [inline R/combineStrata.R:27]: If the input levels is character (or NULL) I would not throw error:
just return(list()) if length(levels) == 0 and let purrr::map handle the rest (for length 1 arguments it works fine

**Review by catalamarti — CHANGES_REQUESTED** [review]
@KimLopezGuell thank you very much, very minor things

**Comment by catalamarti** [comment]
hi @KimLopezGuell I've made NA to work, I don't think it is omopgenerics to judge if NA is a correct strata value or not. What do you think?

**Review by catalamarti — APPROVED** [review]

**Comment by KimLopezGuell** [comment]
That works for me. Thank you @catalamarti

**Pull request merged**